crates/primitives/class/src/compile.rs

```diff
@@ -342,10 +355,28 @@ mod tests {
     }
 
     #[tokio::test]
-    async fn test_compressed_sierra_contract_class_to_json() {
+    async fn test_compressed_sierra_contract_v_1_1_0_class_to_json() {
+        let provider = SequencerGatewayProvider::starknet_alpha_mainnet();
```

Comment:
so this test depends on the current state of mainnet and the gateway ? shouldn't we mock the provider here?

Reply:
you’re right that relying on the current state of the mainnet and the gateway isn’t ideal. However, this is a temporary solution. Storing the entire class within the project would be too heavy. For testing purposes, we need simple and lightweight class definitions from different versions.

we can also open an issue to create dedicated test data (blocks, classes, etc.) that we can use in the future to further improve the reliability of our tests.

Reply:
let's open an issue yes

Reply:
#230